src/main/java/net/minestom/server/network/packet/server/play/DamageEventPacket.java

```diff
@@ -0,0 +1,48 @@
+package net.minestom.server.network.packet.server.play;
+
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.coordinate.Vec;
+import net.minestom.server.network.NetworkBuffer;
+import net.minestom.server.network.packet.server.ServerPacket;
+import net.minestom.server.network.packet.server.ServerPacketIdentifier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static net.minestom.server.network.NetworkBuffer.*;
+
+/**
+ * See <a href="https://wiki.vg/Protocol#Damage_Event">https://wiki.vg/Protocol#Damage_Event</a> for more info.
+ *
+ * @param targetEntityId ID of the entity being damaged
+ * @param damageTypeId   0 if there is no damage type, otherwise the damage type ID + 1
+ * @param sourceEntityId 0 if there is no source entity, otherwise it is entityId + 1
+ * @param sourceDirectId 0 if there is no direct source. For direct attacks (e.g. melee), this is the same as sourceEntityId. For indirect attacks (e.g. projectiles), this is the projectile entity id + 1
+ * @param sourcePos      null if there is no source position, otherwise the position of the source
+ */
+public record DamageEventPacket(int targetEntityId, int damageTypeId, int sourceEntityId, int sourceDirectId, @Nullable Point sourcePos) implements ServerPacket {
```

Comment:
shouldn't you send the packet somewhere?
Ex: LivingEntity's use of EntityAnimationPacket

Reply:
i need help with this

Reply:
in LivingEntity, when shouldAnimate is checked, you should send the new HitAnimationPacket instead of EntityAnimationPacket because TAKE_DAMAGE in that packet is now obsolete
you have to be able to figure out who the "attacker" of the damage is, either with `instanceof` with `EntityDamage` and `EntityProjectileDamage` or adding a new method like `getAttacker` to `Damage`, in order to figure out the correct yaw difference
there is a little bit of random variation in the yaw difference as well, you can see an example in `net.minestom.demo.PlayerInit`

for DamageEventPacket, you can send that packet right before the HitAnimationPacket
you have access to the entity id and damage type id easily
once again, you can do `instanceof` with `EntityDamage` and `EntityProjectileDamage` or you can add `getAttacker` and `getSource` to Damage which are nullable

I think that's the only place